### PR TITLE
Setup gometalinter + fix lint

### PR DESCRIPTION
**Pull request by moul**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,8 +31,8 @@ func main() {
 		g.Error(err, "reading input")
 	}
 
-	if err := proto.Unmarshal(data, g.Request); err != nil {
-		g.Error(err, "parsing input proto")
+	if err2 := proto.Unmarshal(data, g.Request); err2 != nil {
```
Comment (abronan) [inline main.go]: I think the problem here was shadowing, but no need to declare an `err2`. I think you could just as well do:

```
if err = proto.Unmarshal(data, g.Request); err != nil {
```

Just don't use the declaration and assignment sign `:=` there but `=` instead to reuse the first error variable declared above.

Comment (moul) [reply main.go]: > I think the problem here was shadowing

Yes it was

I confess I didn't look enough if the first `err`s were used later in the function or not, will check more properly

File: encoder.go [hunk 2]
```diff
@@ -170,9 +179,9 @@ func (e *GenericTemplateBasedEncoder) Files() []*plugin_go.CodeGeneratorResponse
 	resultChan := make(chan *plugin_go.CodeGeneratorResponse_File, length)
 	for _, templateFilename := range templates {
 		go func(tmpl string) {
-			content, translatedFilename, err := e.buildContent(tmpl)
-			if err != nil {
-				errChan <- err
+			content, translatedFilename, err2 := e.buildContent(tmpl)
```
Comment (moul) [inline encoder.go]: Not sure about how to do it properly, should I reuse the first `err` and create empty `var` variables for `content` and `translatedFilename` to switch the `:=` with `=` or is it better to keep `err2` in this case?

cc @abronan

Comment (vgheri) [reply encoder.go]: Simply doing `content, translatedFilename, err := ... ` should just work, but maybe creating a new variable (like you did) that doesn't escape the closure is cleaner ?

Comment (moul) [reply encoder.go]: using `content, translatedFilename, err := ...` triggers a variable shadowing lint error, I'm declaring variable and reusing previous `err`, it fixes the lint warns, keeps the code readable, and works

**Pull request merged**